### ISSUE TITLE
Use java.util.List for lists

### ISSUE THREAD

**Issue by lorenzleutgeb** [kobylynskyi/graphql-java-codegen#185]
TL;DR: I think that the implementation of this generator misinterprets the semantics of lists in GraphQL. It generates `Collection<T>` where it should generate `List<T>`.

There's mismatch in what classes generated by the library *are expected to provide* (lists) and what they *actually provide* (collections).

### Steps to Reproduce

Generate code for

```graphql
type Foo { bar: [String] }
```

### Expected Result

The Java equivalent of `bar` should have type `List<String>`, and this should of course generalize.

### Actual Result

The Java equivalent of `bar` has type `Collection<String>`.

### Semantics of Lists in GraphQL

[The GraphQL specification (draft)](https://spec.graphql.org/draft/#sec-Type-System.List) considers a list to be
> List values are serialized as **ordered lists**, where each item in the list is serialized as per the item type.

(emphasis mine) which clearly references ordering of elements as a characteristic property of values of the list type.

Also there is [other tooling out there, like Hasura](https://hasura.io/docs/1.0/graphql/manual/queries/sorting.html#sorting-objects) which clearly rely on the property of lists being ordered.

### Semantics of Lists in this generator

The `Collection` type in Java, however, does not guarantee ordering of elements (consider `HashSet` as a counterexample).

**Comment by kobylynskyi** [comment]
@lorenzleutgeb thanks for finding and fixing this!
It's good that I haven't released 2.0.0 yet (planned to do it yesterday, but did not have time). Hopefully will do today if someone will review 179 (maybe you could?)
